docs/preparing_recipes_for_conan2.md

```diff
@@ -0,0 +1,375 @@
+        self.cpp_info.filenames["cmake_find_package_multi"] = "glew" # generates glew-config.cmake
+        ...
+```
+
+In this case we have to set the ``cmake_find_mode`` property for the
+[CMakeDeps](https://docs.conan.io/en/latest/reference/conanfile/tools/cmake/cmakedeps.html#properties)
+generator with value ``both``. That will make CMakeDeps generator create both module and
+config files for consumers (by default it generates just config files).
+
+```python
+class GlewConan(ConanFile):
+    name = "glew"
+    ...
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "GLEW"
+        self.cpp_info.names["cmake_find_package_multi"] = "GLEW"
+        self.cpp_info.filenames["cmake_find_package"] = "GLEW"
+        self.cpp_info.filenames["cmake_find_package_multi"] = "glew"
+
+        self.cpp_info.set_property("cmake_target_name", "GLEW::GLEW")
+
+        self.cpp_info.set_property("cmake_file_name", "GLEW") # generates FindGLEW.cmake
+        self.cpp_info.set_property("cmake_module_file_name", "glew") # generates glew-config.cmake
+
+        # generate both modules and config files
+        self.cpp_info.set_property("cmake_find_mode", "both") 
+        ...
+```
+
+### Understanding some workarounds with the .names attribute model in recipes
+
+The ``.names`` model has some limitations. Because of this, there are some recurrent
+workarounds in recipes to achieve things like setting absolute names for targets (without
+the ``::`` namespace), or for setting a custom namespace. These workarounds can now be
+undone with the ``set_property`` model because it allows setting arbitrary names for CMake
+targets. Let's see some examples of these workarounds in recipes:
+
+* **Use of components to get arbitrary target names in recipes**. Some recipes add a component
+  whose only role is to get a target name that is not limited by the namespaces added by
+  the current generators automatically. For example, the [ktx
+  recipe](https://github.com/conan-io/conan-center-index/blob/5753f954027d9d04b6d05e326f2757ab6b1ac69c/recipes/ktx/all/conanfile.py)
+  uses this workaround to get a target with name ``KTX::ktx``.
+
+```python
+class KtxConan(ConanFile):
+    name = "ktx"
+    ...
+
+    def package_info(self):
+        # changes namespace to KTX::
+        self.cpp_info.names["cmake_find_package"] = "KTX" 
+        ...
+        # the target inherits the KTX:: namespace and sets the target KTX::ktx
+        self.cpp_info.components["libktx"].names["cmake_find_package"] = "ktx"
+        ...
+        # all the information is set via this "fake root" component
+        self.cpp_info.components["libktx"].libs = ["ktx"]
+        self.cpp_info.components["libktx"].defines = [
+            "KTX_FEATURE_KTX1", "KTX_FEATURE_KTX2", "KTX_FEATURE_WRITE"
+        ]
+        ...
+```
+
+In these cases, the recommendation is to add the ``cmake_target_name`` property for both
```

Comment:
Related to this discussion: https://github.com/conan-io/conan-center-index/pull/8428#discussion_r771493541